### ISSUE TITLE
Idea: use minibuffer for input

### ISSUE THREAD

**Issue by alphapapa** [jgkamat/matrix-client-el#29]
@jgkamat It just occurred to me: we could probably just use the minibuffer for input.  "All" we need to do is bind `self-insert-command` in room buffers to a command that reads from the minibuffer and inserts the first character.  That way we wouldn't need to worry about whether point is positioned at the right place to input text, whether the input line has scrolled off-screen, etc.  What do you think?

**Comment by jgkamat** [comment]
Hmm, I have mixed feelings about this idea. I think that it would make it a lot cleaner to insert new lines, but I feel like it's a bit of a context switch to find your typing suddenly in the minibuffer. I think that most packages resolve these issues by making existing text read-only, and defining some handler to jump point to input when any typing is made. In erc, the 'move-to-prompt' module + the 'readonly' module does this: https://github.com/emacs-mirror/emacs/blob/master/lisp/erc/erc-goodies.el#L114.

It seems fairly simple, but it does assume the input line is at point-max:

```
(defun erc-move-to-prompt ()
  "Move the point to the ERC prompt if this is a self-inserting command."
  (when (and erc-input-marker (< (point) erc-input-marker)
             (eq 'self-insert-command this-command))
    (deactivate-mark)
    (push-mark)
    (goto-char (point-max)))
```

I personally wouldn't mind too much if we did it this way, but I could imagine that some people might find this behavior a bit jarring. I think we should see if any potential users feel strongly about it before moving forward (because it would be fairly different from existing chat clients in emacs).
